Find a functionally equivalent code:com.android.internal.util.XmlUtils.readThisListXml:COMMENT
Method Modifier: public      static      final       internal    
Comment:/**
 * Read an ArrayList object from an XmlPullParser.  The XML data could
 * previously have been generated by writeListXml().  The XmlPullParser
 * must be positioned <em>after</em> the tag that begins the list.
 *
 * @param parser The XmlPullParser from which to read the list data.
 * @param endTag Name of the tag that will end the list, usually "list".
 * @param name An array of one string, used to return the name attribute
 * of the list's tag.
 *
 * @return HashMap The newly generated list.
 *
 * @see #readListXml
 */

@@ -1,18 +1,3 @@
 {
-    ArrayList list = new ArrayList();
-    int eventType = parser.getEventType();
-    do {
-        if (eventType == parser.START_TAG) {
-            Object val = readThisValueXml(parser, name);
-            list.add(val);
-        // System.out.println("Adding to list: " + val);
-        } else if (eventType == parser.END_TAG) {
-            if (parser.getName().equals(endTag)) {
-                return list;
-            }
-            throw new XmlPullParserException("Expected " + endTag + " end tag at: " + parser.getName());
-        }
-        eventType = parser.next();
-    } while (eventType != parser.END_DOCUMENT);
-    throw new XmlPullParserException("Document ended before " + endTag + " end tag");
+    return readThisListXml(parser, endTag, name, null);
 }

